scio-core/src/test/scala/com/spotify/scio/values/SCollectionWithSideInputTest.scala

```diff
@@ -379,4 +379,16 @@ class SCollectionWithSideInputTest extends PipelineSpec {
       s should forAll[(Int, Seq[Int])](t => (1 to t._1).map(_ * 2).toSet == t._2.toSet)
     }
   }
+
+  it should "support collect" in {
+    runWithContext { sc =>
+      val p1 = sc.parallelize(Seq("a", "b", "d"))
+      val p2 = sc.parallelize(sideData).asMapSingletonSideInput
+      val pf: PartialFunction[(String, SideInputContext[String]), Int] = {
+        case (i, c) if c(p2).contains(i) => c(p2)(i) * 2
+      }
+      val s = p1.withSideInputs(p2).collect(pf).toSCollection
```

Comment:
🎨 could be written in a slightly more idiomatic way, like:

```suggestion
      val s = p1
        .withSideInputs(p2)
        .collect {
          case (record, ctx) if ctx(p2).contains(record) => ctx(p2)(record) * 2
        }
        .toSCollection
```